anxcloud/resource_object_storage_tenant.go

```diff
@@ -0,0 +1,217 @@
+
+	remoteID := d.Get("remote_id").(string)
+	tenant.RemoteID = &remoteID
+
+	// Set common fields
+	setObjectStorageCommonFields(&tenant, d)
+	setObjectStorageStateField(&tenant, d)
+
+	if err := a.Create(ctx, &tenant); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(tenant.Identifier)
+	return resourceObjectStorageTenantRead(ctx, d, m)
+}
+
+func resourceObjectStorageTenantRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	a := apiFromProviderConfig(m)
+
+	tenant := objectstoragev2.Tenant{Identifier: d.Id()}
+	err := a.Get(ctx, &tenant)
+
+	if api.IgnoreNotFound(err) != nil {
+		return diag.FromErr(err)
+	} else if err != nil {
+		d.SetId("")
+		return nil
+	}
+
+	// Set computed and mutable fields
+	if err := d.Set("name", tenant.Name); err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	if err := d.Set("description", tenant.Description); err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	if err := d.Set("user_name", tenant.UserName); err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	if err := d.Set("backend", tenant.Backend.Identifier); err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+
+	if tenant.Quota != nil {
+		if err := d.Set("quota", *tenant.Quota); err != nil {
+			diags = append(diags, diag.FromErr(err)...)
+		}
+	}
+	if tenant.Usage != nil {
+		if err := d.Set("usage", *tenant.Usage); err != nil {
+			diags = append(diags, diag.FromErr(err)...)
+		}
+	}
+	if tenant.RemoteID != nil {
+		if err := d.Set("remote_id", *tenant.RemoteID); err != nil {
+			diags = append(diags, diag.FromErr(err)...)
+		}
+	} else {
+		// This shouldn't happen since remote_id is required, but handle gracefully
+		if err := d.Set("remote_id", ""); err != nil {
+			diags = append(diags, diag.FromErr(err)...)
+		}
+	}
+
+	// Set common fields
+	setObjectStorageCommonFieldsFromAPI(&tenant, d, &diags)
+	setObjectStorageStateFieldFromAPI(&tenant, d, &diags)
+
+	return diags
+}
+
+func resourceObjectStorageTenantUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	a := apiFromProviderConfig(m)
+
+	tenant := objectstoragev2.Tenant{
+		Identifier:  d.Id(),
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		UserName:    d.Get("user_name").(string),
+		Password:    d.Get("password").(string),
+		Backend: common.PartialResource{
+			Identifier: d.Get("backend").(string),
+		},
+	}
+
+	quota := d.Get("quota").(float64)
+	tenant.Quota = &quota
+
+	remoteID := d.Get("remote_id").(string)
+	tenant.RemoteID = &remoteID
+
+	// Set common fields
+	setObjectStorageCommonFields(&tenant, d)
+	setObjectStorageStateField(&tenant, d)
+
+	// Debug: Print tenant being updated
+	fmt.Printf("DEBUG PROVIDER: About to update tenant\n")
+	fmt.Printf("  ID: %s\n", tenant.Identifier)
+	fmt.Printf("  Name: %s\n", tenant.Name)
+	fmt.Printf("  Description: %s\n", tenant.Description)
```

Comment:
likely leftovers, I think there's some logging inside the terraform framework for debug logs.